kyo-core/shared/src/main/scala/kyo/envs.scala

```diff
@@ -3,45 +3,61 @@ package kyo
 import kyo.core.*
 
 class Envs[+V] extends Effect[Envs[V]]:
-    type Command[T] = Unit
+    type Command[T] = Tag[T]
 
 object Envs:
     private case object envs extends Envs[Any]
     private def envs[V]: Envs[V] = envs.asInstanceOf[Envs[V]]
 
-    def get[V](using Tag[Envs[V]]): V < Envs[V] =
-        envs[V].suspend[V](())
+    // Envs.get[String] would normally use a tag of Envs[String]
+    // Yet we may provide a Layer with an output of String & Int.
+    // This would usually give the handler a tag of Envs[String & Int]
+    // However, then handler.tag != kyo.tag in the handle implementation.
+    // So we have to use this type erased tag and rely on our `accepts` implementation
+    private trait EnvsErased
+    private val envsTag: Tag[EnvsErased]   = Tag[EnvsErased]
+    private inline given [V]: Tag[Envs[V]] = envsTag.asInstanceOf[Tag[Envs[V]]]
+
+    def get[V](using tag: Tag[V]): V < Envs[V] =
+        envs[V].suspend[V](tag)
+
+    inline def run[V: Tag, T: Flat, S, VS, VR](env: V)(value: T < (Envs[VS] & S))(
+        using HasEnvs[V, VS] { type Remainder = VR }
+    ): T < (S & VR) =
+        envs[V].handle(handler[V])(TypeMap(env), value).asInstanceOf[T < (S & VR)]
+    end run
+
+    inline def runTypeMap[V, T: Flat, S, VS, VR](env: TypeMap[V])(value: T < (Envs[VS] & S))(
+        using HasEnvs[V, VS] { type Remainder = VR }
+    ): T < (S & VR) =
+        envs[V].handle(handler[V])(env, value).asInstanceOf[T < (S & VR)]
+    end runTypeMap
 
     class UseDsl[V]:
         inline def apply[T, S](inline f: V => T < S)(
-            using inline tag: Tag[Envs[V]]
+            using
+            inline intersection: Tag.Intersection[V],
+            inline tag: Tag[V]
         ): T < (Envs[V] & S) =
-            envs[V].suspend[V, T, S]((), f)
+            envs[V].suspend[V, T, S](tag, f)
+        end apply
     end UseDsl
 
-    def use[V >: Nothing]: UseDsl[V] =
+    def use[V]: UseDsl[V] =
```

Comment:
hmm I think adding `>: Nothing` here had fixed an issue. Was this change necessary to fix a compilation error?

Reply:
Oh, really? No, I just removed it because `Nothing >: Nothing`, so I think it's unnecessary. But Scala be weird sometimes.

Reply:
Yeah, I did notice it changes the type inference in some cases where `Nothing` would be inferred.

Reply:
Wow! Yeah, adding this back fixed it 😅. Oh, Scala.